### ISSUE TITLE
example code in collections/iface doesn't work

### ISSUE THREAD

**Issue by coaljoe** [zielmicha/collections.nim#3]
https://github.com/zielmicha/collections.nim/blob/cd8ceaa3f611803df4450d41fd1e33bf6c7ce4d5/collections/iface.nim

```
# Define interface type Duck
type Duck = distinct Interface

interfaceMethods Duck:
  quack(number: int): string

# Make implementation of Duck interface
# Implementation objects have to inherit from RootRef.
type AmericanDuck = ref object of RootObj
  name: string

proc quack(d: AmericanDuck, number: int): string =
  return "duck " & d.name & " quacks " & $number & " times"

let myDuck: AmericanDuck = AmericanDuck(name: name)
let myDuckAsInterface: Duck = myDuck.asDuck # or myDuck.asInterface[Duck]

```
gives an error:
/home/k/.nimble/pkgs/collections-0.3.3/collections/iface.nim(121) interfaceMethods
collections.nim(11, 18) template/generic instantiation from here
/home/k/.nimble/pkgs/collections-0.3.3/collections/iface.nim(121, 37) Error: index out of bounds

my other code that uses collections/iface doesn't work either. it worked with previous versions of the library.
im using nim 0.18.0

**Comment by zielmicha** [comment]
Most likely this is due to changes in 0.18.0 (I haven't changed my projects using collections to 0.18.0 yet, they are still 0.17.3). I will investigate this soon.

**Comment by zielmicha** [comment]
This was actually fixed in 77563c623939 in December, the change wasn't released. I just did it.
